vcs/gitcmd/repo.go

```diff
@@ -1053,12 +1053,12 @@ func (fs *gitFSCmd) lsTree(path string) ([]os.FileInfo, error) {
 			mode = mode | vcs.ModeSubmodule
 			cmd := exec.Command("git", "config", "--get", "submodule."+name+".url")
 			cmd.Dir = fs.dir
-			out, err := cmd.CombinedOutput()
-			if err != nil {
-				return nil, fmt.Errorf("invalid `git config` output: %q", out)
+			url := "" // url is not available if submodules are not initialized
+			if out, err := cmd.CombinedOutput(); err == nil {
```

Comment:
I think this should be `cmd.Output` here. You're not parsing the output for errors, only using it when `err == nil`. But `Stderr` may contain extraneous output (info, warnings), which would then become a part of the parsed `url`.

This is related to an overarching issue #33.